core/src/test/scala/org/apache/spark/storage/MemoryStoreSuite.scala

```diff
@@ -291,11 +291,11 @@ class MemoryStoreSuite
     blockInfoManager.removeBlock("b3")
     putIteratorAsBytes("b3", smallIterator, ClassTag.Any)
 
-    // Unroll huge block with not enough space. This should fail and kick out b2 in the process.
+    // Unroll huge block with not enough space. This should fail.
     val result4 = putIteratorAsBytes("b4", bigIterator, ClassTag.Any)
     assert(result4.isLeft) // unroll was unsuccessful
     assert(!memoryStore.contains("b1"))
-    assert(!memoryStore.contains("b2"))
+    assert(memoryStore.contains("b2"))
```

Comment:
FWIW while changing this test to use UnifiedMemoryManager, for a different purpose, I also found this block b2 would not be evicted. I think the assertion maybe isn't correct, so I think this change is OK.